src/ethereum/EthereumEngine.ts

```diff
@@ -121,8 +121,7 @@ export class EthereumEngine extends CurrencyEngine<
     this.networkInfo = env.networkInfo
     this.ethNetwork = new EthereumNetwork(this)
     this.lastEstimatedGasLimit = {
-      publicAddress: '',
-      contractAddress: '',
+      toAddress: '',
```

Comment:
I'll revert this change. We want the publicAddress to trigger a recalculation/call to eth_estimateGas because it may be the case that the destination of the token change may impact the gas limit (the contract may do for different accounts).